apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -0,0 +1,162 @@
+import { showToast } from '~/utils/layout'
+
+interface BuyModalProps extends Omit<BottomModalProps, 'Content'> {}
+
+const BuyModal = (props: BuyModalProps) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<NavigationProp<InWalletTabsParamList>>()
+  const theme = useTheme()
+  const webViewRef = useRef<WebView>(null)
+  const insets = useSafeAreaInsets()
+  const defaultAddress = useAppSelector(selectDefaultAddress)
+  const allAddresses = useAppSelector(selectAllDiscoveredAddresses)
+  const [isDisclaimerAccepted, setIsDisclaimerAccepted] = useState(false)
+
+  const onAndroidBackPress = () => {
+    if (webViewRef.current) {
+      webViewRef.current.goBack()
+      return true // prevent default behavior (exit app)
+    }
+    return false
+  }
+
+  useEffect(() => {
+    if (Platform.OS === 'android') {
+      BackHandler.addEventListener('hardwareBackPress', onAndroidBackPress)
+      return () => {
+        BackHandler.removeEventListener('hardwareBackPress', onAndroidBackPress)
+      }
+    }
+  }, [])
+
+  const receivingAddress = defaultAddress?.hash || allAddresses[0]?.hash
+
+  if (!receivingAddress) {
+    showToast({
+      text1: t('Could not find a receiving address'),
+      text2: t('Cannot proceed because no address has been found to receive the funds.'),
+      type: 'error'
+    })
+    return null
+  }
+
+  const banxaURL =
+    'https://alephium.banxa-sandbox.com/' +
+    `?walletAddress=${receivingAddress}` +
+    `&theme=${theme.name}` +
+    `&backgroundColor=${theme.bg.primary.slice(1)}` +
+    `&textColor=${theme.font.primary.slice(1)}` +
+    `&primaryColor=${theme.global.accent.slice(1)}` +
+    `&secondaryColor=${theme.global.complementary.slice(1)}`
+
+  const handleModalCloseWhenFinished = (e: WebViewNavigation) => {
+    // Close modal if url equals default return URL configured in Banxa dashboard
+    if (e.url.includes('alephium.org')) {
+      navigation.navigate('TransfersScreen')
+      props.onClose()
+    }
+  }
+
+  return (
+    <Portal>
+      <BottomModal
+        {...props}
+        Content={(props) => (
+          <ModalContent {...props} contentContainerStyle={{ flex: 1, paddingTop: 0 }}>
+            {!isDisclaimerAccepted && (
+              <DisclaimerContent>
+                <ScreenTitle title={t('Disclaimer')} />
+                <AppText style={{ flex: 1 }}>
+                  <Trans
+                    t={t}
+                    i18nKey="banxaDisclaimer"
+                    components={{
+                      1: <LinkToWeb url="https://www.banxa.com" text="Banxa.com" />
+                    }}
+                  >
+                    {
+                      'You are about to access 3rd party services provided by <1/> through an in-app browser. Alephium does not control Banxa’s services. Banxa’s terms and conditions will apply, so please read and understand them before proceeding.'
+                    }
+                  </Trans>
```

Comment:
This might cause confusion to our translators because they have no context as to what the `[...] by <1/> through [...]` is. Is it a link? A name of a company? A physical person? Each of these might have different articles and prepositions based on the language. Normally, in these cases, we would have `[...] by <1>Banxa.com</1> through [...]`. We can enhance the translators experience and avoid wrong articles and prepositions by making a small adjustment to the `LinkToWeb` component to accept `children: string` instead of `text: string` and updating to

```suggestion
                  <Trans
                    t={t}
                    i18nKey="banxaDisclaimer"
                    components={{
                      1: <LinkToWeb url="https://www.banxa.com" />
                    }}
                  >
                    {
                      'You are about to access 3rd party services provided by <1>Banxa.com></1> through an in-app browser. Alephium does not control Banxa’s services. Banxa’s terms and conditions will apply, so please read and understand them before proceeding.'
                    }
                  </Trans>

Reply:
Yep, I hesitated. I'll do it.

Reply:
It actually doesn't work 🤔 Child doesn't seem to be passed to the translated component.

<img width="360" alt="image" src="https://github.com/user-attachments/assets/2206d24c-8328-41d6-b633-296ac33a17c2">

Reply:
Done